llvm_util/utils.cpp

```diff
@@ -41,10 +41,34 @@ ostream *out;
 
 const llvm::DataLayout *DL;
 
+static void _getUsedGlobalVariables(llvm::User *U,
+    llvm::SmallPtrSet<llvm::Value *, 16> &Visited,
+    llvm::SmallPtrSet<llvm::GlobalVariable *, 16> &GlobalVars) {
+  Visited.insert(U);
+  for (auto I = U->op_begin(), E = U->op_end(); I != E; ++I) {
+    llvm::Value *V = I->get();
+    if (Visited.count(V)) continue;
+    else if (llvm::GlobalVariable *GV = dyn_cast<llvm::GlobalVariable>(V)) {
+      GlobalVars.insert(GV);
+    }
+  }
+}
+
 }
 
 namespace llvm_util {
 
+vector<llvm::GlobalVariable *> getUsedGlobalVariables(llvm::Function &F) {
+  llvm::SmallPtrSet<llvm::Value *, 16> Visited;
+  llvm::SmallPtrSet<llvm::GlobalVariable *, 16> GlobalVars;
```

Comment:
why are you using LLVM datastructures? We don't want to rely on LLVM stuff whenever it's not strictly needed

Reply:
updated the code to use std::set.